ktor-features/ktor-html-builder/jvm/src/io/ktor/html/RespondHtml.kt

```diff
@@ -32,9 +33,18 @@ public class HtmlContent(
         get() = ContentType.Text.Html.withCharset(Charsets.UTF_8)
 
     override suspend fun writeTo(channel: ByteWriteChannel) {
-        channel.bufferedWriter().use {
-            it.append("<!DOCTYPE html>\n")
-            it.appendHTML().html(block = builder)
+        val content = ByteArrayOutputStream()
+
+        try {
+            content.bufferedWriter().use {
+                it.append("<!DOCTYPE html>\n")
+                it.appendHTML().html(block = builder)
+            }
+        } catch (cause: Throwable) {
+            channel.close(cause)
```

Comment:
Don't we already do that on the call side of the `writeTo`?

Reply:
Nope, but we probably should